### ISSUE TITLE
cinnamon general settings minimum memory limit of 1024MB too high for many systems

### ISSUE THREAD

**Issue by dhdurgee** [linuxmint/cinnamon#10287]
System information in attached file.
[inxi.log](https://github.com/linuxmint/cinnamon/files/7040137/inxi.log)

I have an older system with 8GB of memory.  When started cinnamon is using about 70MB, so a 1024MB limit is far too high for this system.
[cinnamon_monitor.zip](https://github.com/linuxmint/cinnamon/files/7040162/cinnamon_monitor.zip)

Until this lower limit is relaxed I will need to continue using the script in the attached zip file to keep cinnamon memory usage under control.

Feel free to leave the default where it is, but please reduce the minimum limit to 256MB.

**Comment by dhdurgee** [comment]
I believe I have located the file that must be modified to adjust this minimum:

/usr/share/cinnamon/cinnamon-settings/modules/cs_general.py

[cs_general.zip](https://github.com/linuxmint/cinnamon/files/7040464/cs_general.zip)

I have attached a zip file containing this file.  Unless I am mistaken I believe all that is needed is to change "1024" to "256" in line 49 of this file.

**Comment by leigh123linux** [comment]
-1 to setting it low, I don't want cinnamon to restart every hour.
The cerrent setting is fine as it never hits the limit.

**Comment by dhdurgee** [comment]
I thank you both for your timely input on this issue.

I appreciate the pointer to dconf-editor as a means to address this issue.  I should note that this is not installed by default.

Please also note that I was not suggesting that the default value be set low, I was simply suggesting that the current minimum acceptable value in the cinnamon general settings GUI for this value is too restrictive in cases such as mine.  Perhaps it would be most appropriate for the default value to be a multiple of the memory initially allocated by cinnamon at startup as opposed to a fixed value.

In my case cinnamon at startup is using less than 70MB of memory as reported by the system monitor.  Obviously this will vary dramatically depending upon the user's configuration and usage of cinnamon.  The current minimum permissible value that can be set via the GUI at present is about 15 times the memory that cinnamon uses at startup.  I believe that 4-5 times the initial memory allocation to be more appropriate.

At present I am running the script in my initial post which I developed after problems with run-away memory usage encountered in the mint 20.1 cinnamon release.  This script is still working well with the current release, but I would be fine with discontinuing its use and using the built-in capabilities instead.

Obviously this setting needs to be tuned to fit the circumstances.  The settings I have in this script thus far have resulted in one restart after 23:36 of operation on this system.

One difficulty I have in migrating from this script would be the question of what setting for this limit matches that in the script, as I have found that system monitor and ps appear to report differing numbers.  Given this I would imagine that whatever number the limit compares to will likely also differ.

If I do not patch the file I mentioned in my second post and make use of dconf-editor to modify the value will the cinnamon settings GUI respect this?  I also assume that a cinnamon restart will be required for it to be in force, correct?

Once again I thank you for your participation in this issue.

I would appreciate participation in the two other issues I have raised that are still outstanding:

https://github.com/linuxmint/cinnamon/issues/10196

and

https://github.com/linuxmint/cinnamon/issues/10092

**Comment by dhdurgee** [comment]
file:///home/dhdurgee/Downloads/restarts.zip

Decided to give your feature a try, so I used dconf-editor to set the value to 320, disabled my script in the startup list and restarted my system. All still looked good when I went to bed last night. Unfortunately I woke up to a surprise.

When I came to unlock the system this morning I was faced with a black screen saying something about a screensaver failure and the need to use Ctrl-Alt-F1, login and issue a command to unlock the keyboard. Interestingly enough the Ctrl-Alt-F1 session would not open either, the mouse cursor simply disappeared. Given this I tried Ctrl-Alt-F2 and was able to get the task done.

I have attached a zip file containing what I think to be the relevant logs. If I am reading this correctly the restart was triggered at 01:32:56 this morning, but cinnamon was not actually restarted until 07:47:36 when I unlocked the keyboard from terminal 2!

Exactly what happened is a mystery to me. I am now debating trying one more time with your feature or disabling it and going back to my script. I never had such a problem with my approach. Maybe this was a fluke, maybe your restart approach is flawed.  My method of restart is simply to automate the restart cinnamon hotkey via xdotool and appears to work fine.

**Comment by dhdurgee** [comment]
Decided to give your feature one more chance and got another improper restart that required me to switch to terminal 2 and issue the "cinnamon-unlock-desktop" command.  I am going back to my script until you fix this.  I am attaching this morning's logs for your analysis.
[restarts.zip](https://github.com/linuxmint/cinnamon/files/7072454/restarts.zip)

**Comment by torsumy** [comment]
becauser cinnamon is so heavy and shitty laggy

**Comment by jondo** [comment]
(I guess that the issue described by @dhdurgee is https://github.com/linuxmint/cinnamon-screensaver/issues/392.)

**Comment by dhdurgee** [comment]
Yes, that is the issue.  I have installed the patch and no longer see the crash.  After a bit of research I was able to determine that the multicore system monitor was leaking memory and replaced it with another system monitor.  With the new monitor in place and using dconf-editor to set it to 192 I have running for almost a week and cinnamon has yet to restart.

So once again I would argue that the 1024 minimum setting is far to high.

**Comment by dhdurgee** [comment]
I have now been running with a 192 limit for several weeks.  After changing out an applet that leaked memory I now find that I see a restart every eight days.

So once again I argue that the hard lower limit in the GUI settings should be reduced from 1024 and given my experience 256 might be a reasonable alternative.  Obviously someone determined can bypass the GUI as I have here, but why require it?
